content/kotlin/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,383 @@
+val b: Double = sqrt(a)
+val c: Double = a.pow(2)
+
+println("The square root of $a is $b.")
+println("$a raised to the power of 2 is $c.")
+
+}
+```
+
+The output will be:
+
+```
+The square root of 16.0 is 4.0.
+16.0 raised to the power of 2 is 256.0.
+```
+
+### Float
+
+In Kotlin, the `Float` data type represents a 32-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
+
+Here is an example of how to declare a Float variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Float = 3.14f
+    val b: Float = 2.71f
+    val c: Float = a * b as Float
+    print(c)
+
+}
+```
+
+The output will be :
+
+```
+8.5094
+```
+
+In this example, the c variable will be assigned the value 8.5014, which is the result of 3.14 \* 2.71.
+
+### Int
+
+In Kotlin, the `Int` data type represents a 32-bit signed integer. It is a primitive data type that can hold values ranging from -2147483648 to 2147483647.
+
+Here is an example of how to declare an `Int` variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Int = 50
+    val b: Int = 25
+    val c: Int = a + b
+    print(c)
+}
+```
+
+The output of the above code will be:
+
+```
+75
+```
+
+### Long
+
+In Kotlin, the `Long` data type represents a 64-bit signed integer. It is a primitive data type that can hold values ranging from -9223372036854775808 to 9223372036854775807.
+
+Here is an example of how to declare a `Long` variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Long = 500000000
+    val b: Long = 250000000
+    val c: Long = a + b
+    print(c)
+}
+```
+
+The output for the above code will be :
+
+```
+750000000
+```
+
+In this example, the c variable will be assigned the value 750000000, which is the result of 500000000 + 250000000.
+
+### Short
+
+In Kotlin, the `Short` data type represents a 16-bit signed integer. It is a primitive data type that can hold values ranging from -32768 to 32767.
+
+Here is an example of how to declare a `Short` variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Short = 50
+    val b: Short = 25
+    val c: Short = (a + b).toShort()
+    print(c)
+}
+```
+
+The output for the above code will be:
+
+```
+75
+```
+
+In this example, the c variable will be assigned the value 75, which is the result of 50 + 25. The `toShort()` function is used to explicitly convert the result to a `Short` type.
+
+## Non-Primitive Data Types
+
+The following are non-primitive data types in Kotlin :
+
+- Any
+- Arrays
+- Class
+- Enum
+- List
+- Map
+- Nothing
+- Set
+- String
+- Unit
+
+### Arrays
+
+In Kotlin, an `Array` is a data structure that stores a fixed-size collection of elements of the same data type. Arrays are non-primitive data types, which means they are derived from primitive data types or other non-primitive data types.
+
+To declare an Array in Kotlin, the `arrayOf()` function can be utilized, with the elements of the array specified within the parentheses. For instance:
```

Comment:
@yangc95 I removed "you" and changed the sentence like this check if this is right?